kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactory.java

```diff
@@ -71,7 +71,7 @@
 
     private final AtomicInteger transactionIdSuffix;
 
-    private volatile CloseLazyProducer<K, V> producer;
+    private volatile CloseNeverProducer<K, V> producer;
```

Comment:
This `volatile` `Producer` is specific for the case we are in `ConfirmationMode#WAIT_FOR_ACK` or `ConfirmationMode#NONE`. 
I'd suggest to rename this field to `nonTransactionalProducer` to be a bit more specific.

Reply:
You're absolutely right, that adds a lot of clarity.